app/src/main/java/ru/netology/custom_view/ui/StatsView.kt

```diff
@@ -110,43 +130,84 @@ class StatsView @JvmOverloads constructor(
     }
 
     override fun onDraw(canvas: Canvas) {
-        if (data.isEmpty()) {
+        if (data.isEmpty() || progress == 0F) {
             return
         }
+        var startFrom = INITIAL_ANGLE
+        val maxAngle = 360 * progress + INITIAL_ANGLE
 
-        var startFrom = -90F
-        paint.color = colorOfEmptySpace
-        canvas.drawCircle(center.x, center.y, radius, paint)
+        drawCircleBackground(canvas)
 
         for ((index, datum) in data.withIndex()) {
 
-            val angle = if (sumValues > 100F) {
-                360F * (datum / sumValues)
-            } else {
-                (datum * 3.6F)
+            val angle = 360 * (datum / sumValues)
+            val sweepAngle = min(angle, maxAngle - startFrom)
+
+            if (startFrom + angle > maxAngle) {
+                drawData(
+                        index = index,
+                        canvas = canvas,
+                        startFrom = startFrom,
+                        sweepAngle
+                )
+                return
             }
```

Comment:
А если так попробовать? 😉
```suggestion
            if (startFrom > maxAngle) return
            
            val angle = 360 * (datum / sumValues)
            val sweepAngle = min(angle, maxAngle - startFrom)
```

Reply:
да, так намного лаконичней код получается, а поведение отрисовки то же)